platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/ProblemDefinition.java

```diff
@@ -72,17 +70,4 @@ public interface Problem {
      * A list of possible solutions the user can try to fix the problem.
      */
     List<String> getSolutions();
```

Comment:
❓ Same as with `getSeverity`: Does this mean that the same category with different solutions is considered a completely separate issue?

I could agree with the `getSeverity()` case, but not with this... If you think about it, the more contextual a solution, the better it is.

Reply:
That's right, thanks for thinking hard about this change. To phrase it differently, solutions probably/mostly will contain context- dependent advice.

Reply:
There are a bunch of properties on the ProblemReport interface that should have a context sensitive and a context insensitive variant.

Reply:
There are cases where there's a general advice you can supply with the problem definition. At the same time, there's 100% need for a solution within a context. I think I'll leave it like this, and introduce ProblemContext.getSolutions, (and the corresponding ProblemSpec.contextualSolution()) separately. Let me know if you disagree.

Reply:
I think the definition/context split might be a good solution.